src/main/java/Duke.java

```diff
@@ -1,10 +1,149 @@
+import java.util.Scanner;
+
+
 public class Duke {
+
+    public static int listSize = 0;
+    public static Task[] taskList = new Task[100];
+
+    public static void addTask(Task task) {
+        taskList[listSize] = task;
+        listSize++;
+        System.out.printf("Got it. I've added this task:\n%s\nNow you have %d tasks in the list.\n",
+                task.print(), listSize);
+    }
+
+    public  static  void todo(String line, String[] words) {
+        if (words.length >= 2) {
+            String task = line.substring(line.indexOf(' ') + 1);
+            Todo todo = new Todo(task);
+            addTask(todo);
+        } else {
+            System.out.println("Usage: todo <task>");
+        }
+    }
+
+    public static void deadline(String line) {
+        try {
+            String task = line.substring(line.indexOf(' ')+1, line.indexOf("/by")-1);
+            String date = line.substring(line.indexOf("/by")+4);
+            Deadline deadline = new Deadline(task, date);
+            addTask(deadline);
+        } catch (StringIndexOutOfBoundsException ex) {
+            System.out.println("Usage: deadline <task> /by <date>");
+        }
+    }
+
+    public static void event(String line) {
+        try {
+            String task = line.substring(line.indexOf(' ')+1, line.indexOf("/at")-1);
+            String date = line.substring(line.indexOf("/at")+4);
+            Event event = new Event(task, date);
+            addTask(event);
+        } catch (StringIndexOutOfBoundsException ex) {
+            System.out.println("Usage: event <task> /at <date>");
+        }
+    }
+
+    public static void list() {
+        if (listSize > 0) {
+            System.out.println("Here are the tasks in your list:");
+            for (int i = 0; i < listSize; i++) {
+                System.out.printf("%d. %s\n", i + 1, taskList[i].print());
+            }
+        } else {
+            System.out.println("No items in list! Type something to add to list.");
+        }
+    }
+
+    public static void mark(boolean isMark, String[] words) {
+        if (words.length == 2) {
+            try {
+                int d = Integer.parseInt(words[1]);
```

Comment:
Maybe consider using a better noun instead of a single letter d to represent this as it is ambiguous?